scripts/jira/cloner.py

```diff
@@ -0,0 +1,216 @@
+        if link.type.name != 'Cloners':
+            continue
+        if hasattr(link, 'inwardIssue'):
+            l.append(connection.issue(link.inwardIssue.key))
+    return l
+
+def get_clones_issue(issue, connection):
+    for link in issue.fields.issuelinks:
+        if link.type.name != 'Cloners':
+            continue
+        if hasattr(link, 'outwardIssue'):
+            return connection.issue(link.outwardIssue.key)
+    return None
+
+def set_fix_versions(issue, fixVersions, connection):
+    issue.update(fields={
+        'fixVersions': [{'name': x} for x in fixVersions]
+    })
+
+def clone_issue(issue, target, connection):
+    data_dict = {}
+    data_dict['priority'] = {'id': issue.fields.priority.id}
+    data_dict['labels'] = issue.fields.labels + ['backport']
+    data_dict['assignee'] = {'name': issue.fields.assignee.name}
+    data_dict['reporter'] = {'name': issue.fields.reporter.name}
+    data_dict['issuetype'] = {'name': issue.fields.issuetype.name}
+    data_dict['project'] = {'id': issue.fields.project.id}
+    data_dict['summary'] = issue.fields.summary
+    data_dict['description'] = issue.fields.description
+    data_dict['components'] = [{'id': x.name} for x in issue.fields.components]
+    data_dict['versions'] = [{'name': x.name} for x in issue.fields.versions]
+    data_dict['fixVersions'] = [{'name': x.name} for x in issue.fields.fixVersions]
+    # QA contact
+    if hasattr(issue.fields, 'customfield_12315948'):
+        data_dict['customfield_12315948'] = {'value': issue.fields.customfield_12315948.value}
+    data_dict['customfield_12320947'] = [{'value': x.value} for x in issue.fields.customfield_12320947]
+    # Target version
+    data_dict['customfield_12319940'] = [{'name': target}]
+
+    new_issue = connection.create_issue(data_dict)
+    connection.create_issue_link("Cloners", inwardIssue=new_issue.key, outwardIssue=issue.key)
+    connection.create_issue_link("Blocks", inwardIssue=issue.key, outwardIssue=new_issue.key)
+    return new_issue
+
+def check_clones(issue, connection, dry_run):
+    actions = []
+    try:
+        fix_versions = get_issue_fix_versions(issue)
+    except RuntimeError as e:
+        actions.append([JIRA_URL_PREFIX+issue.key, "", "Fix versions field empty. Please fill it."])
+        return actions
+
+    try:
+        target_version = get_issue_target_versions(issue)
+    except RuntimeError as e:
+        actions.append([JIRA_URL_PREFIX+issue.key, "", f"Target field problems: {e}. Please check."])
+        return actions
+
+    if not is_original_issue(issue):
+        return actions
+
+    if target_version != fix_versions[0]:
+        actions.append([JIRA_URL_PREFIX+issue.key, "", f"Target version does not target first Fix version. {target_version} != {fix_versions[0]}. Please check."])
+        return actions
+
+    fix_versions_missing = set(fix_versions[1:])
+    clones = get_clone_by_issues(issue, connection)
+    for clone in clones:
```

Comment:
The body of this loop has a lot of conditionals and exception handling. We can land this version, but you might be able to simplify this by creating a "command" abstraction so that this function can be split between building a list of commands to run and another function that actually runs them (or not, if dry-run is set).